### PR TITLE
bump github.com/google/gofuzz

### PR DESCRIPTION
**Pull request by dprotaso**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This bump adds the ability for the fuzzer to skip certain struct fields

**Special notes for your reviewer**:
Thanks for the review!

**Release Notes**:
```release-note
NONE
```

### CODE REVIEW

**Comment by matthyx** [comment]
@dprotaso how does it work, do you need to instruct it about which structs to skip?

File: staging/src/k8s.io/apimachinery/go.mod [hunk 1]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/golang/groupcache v0.0.0-20160516000752-02826c3e7903
 	github.com/golang/protobuf v1.3.2
 	github.com/google/go-cmp v0.3.0
-	github.com/google/gofuzz v1.0.0
+	github.com/google/gofuzz v1.0.1-0.20191028174853-db92cf7ae75e
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/go.mod]: unless there's a good reason not to, we prefer to use semver-tagged releases (it helps ensure we are consuming a version the library author intended to be used)

you will also need to bump this by following the instructions at https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/vendor.md#adding-or-updating-a-dependency, which are basically:

```
hack/pin-dependency.sh github.com/google/gofuzz $new-version
hack/update-vendor.sh
<commit>
```

Comment (dprotaso) [reply staging/src/k8s.io/apimachinery/go.mod]: done

**Comment by dprotaso** [comment]
> @dprotaso how does it work, do you need to instruct it about which structs to skip?

@matthyx it's pretty straight forward https://godoc.org/github.com/google/gofuzz#Fuzzer.SkipFieldsWithPattern

**Comment by matthyx** [comment]
> > @dprotaso how does it work, do you need to instruct it about which structs to skip?
> 
> @matthyx it's pretty straight forward https://godoc.org/github.com/google/gofuzz#Fuzzer.SkipFieldsWithPattern

Alright, so you plan to do that for some fields in a follow-up PR?

File: staging/src/k8s.io/apimachinery/go.mod (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/apimachinery/go.mod]: > unless there's a good reason not to, we prefer to use semver-tagged releases (it helps ensure we are consuming a version the library author intended to be used)

This is still outstanding. It doesn't look like there's a tagged version of this library yet that has the changes you want. In this scenario, we usually reach out to the upstream component to ask what the timeframe for a tagged release is.

Comment (dprotaso) [reply staging/src/k8s.io/apimachinery/go.mod]: This snippet is out of date

**Comment by dprotaso** [comment]
> Alright, so you plan to do that for some fields in a follow-up PR?

@matthyx someone pointed out that my reply might have came off as coarse - That wasn't my intention and my apologies.

Back to your question - yeah there's some minor cleanup that can happen here:
https://github.com/kubernetes/kubernetes/blob/f10de54bc04bcc7e726e55a232ff1b1718019a65/staging/src/k8s.io/client-go/rest/config_test.go#L316-L322

An additional use case I've also encountered is for downstream consumers of apimachinery's roundtrip testing package. A custom fuzzer can be provided and being able to skip fields is a scenario I have encountered.

The functions I'm referring to are:
https://github.com/kubernetes/kubernetes/blob/dcd0755f84bf8c15fd21e8d697b924991f641ae4/staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/roundtrip.go#L149-L155

**Comment by matthyx** [comment]
@dprotaso no worries :-)
/ok-to-test
/lgtm

**Comment by dprotaso** [comment]
heyo - following up to see if there are any more changes required?

**Comment by dprotaso** [comment]
thanks @liggitt

**Pull request merged**